doc/en/backwards-compatibility.rst

```diff
@@ -3,6 +3,57 @@
 Backwards Compatibility Policy
 ==============================
 
+.. versionadded: 6.0
+
+pytest is actively being evolved and as a project that is been decades in the making,
+we keep learning about new and better structures to express different details about testing.
+
+While we implement those we try to ensure easy transition and don't want to impose unnecessary churn on our users and community/plugin authors.
+
+We are considering multipe Types of backward compatibility considering transitions
```

Comment:
```suggestion
As of now, pytest considers multipe types of backward compatibility transitions:
```

After all we are not considering changing, we are changing correct?

Reply:
"multiple" as well